Development.md

```diff
@@ -0,0 +1,89 @@
+
```

Comment:
Nit:

Perhaps it'd make more sense to add a `docs/` directory and put this in there under `development-environment.md`. As it stands this file is a little bit of everything :(
Instead it'd be nice if we could get a proper `CONTRIBUTING.md` into the root folder which just has some general contributing guidelines like, where to look for more information, a notice about coding standards or anything else one might want to keep in mind when starting to work on the plugin.